--- a/src/hal/plus/rk_hal.c
+++ b/src/hal/plus/rk_hal.c
@@ -26,6 +26,8 @@
 char _rk_vi_pipe = 0;
 char _rk_vpss_chn = 0;
 char _rk_vpss_grp = 0;
+static void *_rk_uclibc_compat = NULL;
+static void *_rk_rga = NULL;
 
 void rk_hal_deinit(void)
 {
@@ -37,12 +39,33 @@
     rk_aud_unload(&rk_aud);
     rk_aiq_unload(&rk_aiq);
     rk_sys_unload(&rk_sys);
+
+    if (_rk_rga)
+        dlclose(_rk_rga);
+    if (_rk_uclibc_compat)
+        dlclose(_rk_uclibc_compat);
+    _rk_rga = NULL;
+    _rk_uclibc_compat = NULL;
 }
 
 int rk_hal_init(void)
 {
     int ret;
 
+    _rk_uclibc_compat = dlopen("libuclibc-compat.so", RTLD_LAZY | RTLD_GLOBAL);
+    if (!_rk_uclibc_compat) {
+        HAL_ERROR("rk_hal", "Failed to load libuclibc-compat.so!\nError: %s\n", dlerror());
+        return EXIT_FAILURE;
+    }
+
+    _rk_rga = dlopen("librga.so", RTLD_LAZY | RTLD_GLOBAL);
+    if (!_rk_rga) {
+        HAL_ERROR("rk_hal", "Failed to load librga.so!\nError: %s\n", dlerror());
+        dlclose(_rk_uclibc_compat);
+        _rk_uclibc_compat = NULL;
+        return EXIT_FAILURE;
+    }
+
     if (ret = rk_sys_load(&rk_sys))
         return ret;
     if (ret = rk_aiq_load(&rk_aiq))
